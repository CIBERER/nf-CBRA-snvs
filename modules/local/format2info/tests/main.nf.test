// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test format2info
nextflow_process {

    name "Test Process FORMAT2INFO"
    script "../main.nf"
    process "FORMAT2INFO"

    tag "modules"
    tag "modules_nfcore"
    tag "format2info"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("format2info - vcf") {

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("https://zenodo.org/records/15730236/files/NA12878-HG001.hg38.final.chr1.chr22.vcf.gz?download=1", checkIfExists: true),
                    file("https://zenodo.org/records/15730236/files/NA12878-HG001.hg38.final.chr1.chr22.vcf.gz.tbi?download=1.tbi", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }
}