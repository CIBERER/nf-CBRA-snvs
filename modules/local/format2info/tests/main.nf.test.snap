{
    "format2info - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.vcf_to_annotate.vcf.gz:md5,e8a8ebc0fa2506cb53a121ed2c2d970b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fields.txt:md5,261a415884e458e0ef37c68a3cdf426e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,228c36eb4b6c94cbbac8f502502faf05"
                ],
                "fields": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fields.txt:md5,261a415884e458e0ef37c68a3cdf426e"
                    ]
                ],
                "vcf_to_annotate": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.vcf_to_annotate.vcf.gz:md5,e8a8ebc0fa2506cb53a121ed2c2d970b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,228c36eb4b6c94cbbac8f502502faf05"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-24T13:41:09.718560855"
    }
}